client/app/services/parameters/Parameter.js

```diff
@@ -0,0 +1,151 @@
+import { isNull, isObject, isFunction, isUndefined, isEqual, has, omit, isArray, each } from 'lodash';
+import {
+  TextParameter, NumberParameter, EnumParameter, QueryBasedDropdownParameter,
+  DateParameter, DateRangeParameter,
+} from '.';
+
+class Parameter {
+  constructor(parameter, parentQueryId) {
+    this.title = parameter.title;
+    this.name = parameter.name;
+    this.type = parameter.type;
+    this.global = parameter.global; // backward compatibility in Widget service
+    this.parentQueryId = parentQueryId;
+
+    // Used for meta-parameters (i.e. dashboard-level params)
+    this.locals = [];
+
+    // Used for URL serialization
+    this.urlPrefix = 'p_';
```

Comment:
That doesn't concern me that much, but do you have any other idea in mind?

Reply:
Well nothing concrete. My thinking is that `Parameter` should be agnostic to this property because it isn't the one using it. Dashboards, Widgets and Query pages use it with their own logic - therefore they should be the ones executing that logic.

But yeah I'd leave it as is for now :)